### PR TITLE
Mcp 2550 - Adjust MAS BP annotation handling

### PR DESCRIPTION
**Pull request by wlin2019**
Using regex to handling MAS BP annotation handling

## What was the problem?
For some MAS BP annotation data, VRO cannot handle it gracefully, and create some crashing error messages. See MCP-2550.

Associated tickets or Slack threads:

- [MCP-000](https://amida.atlassian.net/browse/MCP-000)

## How does this fix it?[^1]

Using regular expression to handle the BP annotation data. It keeps the original approach to handle the missing BP reading value with default value (0). Consequently, I also adjusted Lighthouse service code to handle missing BP reading value to be consistent with MAS handling. See PR #1293.

## How to test this PR
- Step 1
- Step 2[^secrel]

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by ianwolson — APPROVED** [review]

File: service/provider/src/main/java/gov/va/vro/service/provider/mas/service/mapper/MasCollectionAnnotsResults.java [hunk 1]
```diff
@@ -119,27 +133,52 @@ public AbdEvidence mapAnnotationsToEvidence(MasCollectionAnnotation masCollectio
...
 
-  private static AbdBloodPressure createBloodPressure(MasAnnotation masAnnotation) {
-    String[] bpValues = masAnnotation.getAnnotVal().split("/");
+  private static AbdBloodPressure createBloodPressure(MasAnnotation masAnnotation)
+      throws MasException {
+    log.info("MasAnnotation: {}", masAnnotation.getAnnotVal());
+    // Pattern to handle - "120/80 mg/dL"; // or "120/80" or "-/80" or "120/-"
+    Pattern pattern = Pattern.compile(MAS_BP_READING_VERIFICATION);
+    Matcher matcher = pattern.matcher(masAnnotation.getAnnotVal());
+    if (!matcher.matches()) {
```
Comment (rajesh-Krishnasamy) [inline service/provider/src/main/java/gov/va/vro/service/provider/mas/service/mapper/MasCollectionAnnotsResults.java:142]: Could you please check with the business team to see whether an exception has to be raised if the BP annotation values don't match the expected format.

**Review by au2 — DISMISSED** [review]
Remove LABRESULT from the cases we handle. I verified that we are not receiving those. Or any blood pressure will be send by those.

**Comment by wlin2019** [comment]
> Remove LABRESULT from the cases we handle. I verified that we are not receiving those. Or any blood pressure will be send by those.

Removed.

**Review by alan-amida — APPROVED** [review]
Looks good. Thanks, Warren!

**Review by au2 — CHANGES_REQUESTED** [review]

**Pull request merged**